fastly/resource_fastly_tls_subscription_test.go

```diff
@@ -43,12 +44,16 @@ func TestAccResourceFastlyTLSSubscription(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "managed_dns_challenge.%", "3"),
 					resource.TestCheckResourceAttrSet(resourceName, "managed_http_challenges.#"),
 					resource.TestCheckResourceAttr(resourceName, "common_name", domain1),
-					testAccResourceFastlyTLSSubscriptionExists(resourceName),
+					testAccResourceFastlyTLSSubscriptionExists(resourceName, &subscriptionId),
 				),
 			},
 			{
 				Config: testAccResourceFastlyTLSSubscriptionConfig(name, domain1, domain2, commonName2),
-				Check:  resource.TestCheckResourceAttr(resourceName, "common_name", domain2),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "common_name", domain2),
+					// subscription is "updated" so the subscription ID should remain the same
+					resource.TestCheckResourceAttrPtr(resourceName, "id", &subscriptionId),
```

Comment:
ref. https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/helper/resource#TestCheckResourceAttrPtr
We can't use `TestCheckResourceAttr` here because then it's always an empty string (default value of `string`) and not reflecting the changes made in the previous step.